torch/nn/functional.py

```diff
@@ -3756,6 +3758,13 @@ def multi_head_attention_forward(query,                           # type: Tensor
     q = q * scaling
 
     if attn_mask is not None:
+
+        # convert ByteTensor attn_mask to float
+        if attn_mask.dtype == torch.uint8:
+            attn_mask = torch.zeros(attn_mask.size(),
```

Comment:
Instead of torch.zeros you should be able to use [new_empty](https://pytorch.org/docs/master/tensors.html#torch.Tensor.new_empty).